src/container.jl

```diff
@@ -190,12 +197,34 @@ function update_keys!(pc::ParticleContainer, ref::Union{Particle,Nothing}=nothin
     n = ref === nothing ? nparticles : nparticles - 1
     for i in 1:n
         pi = pc.vals[i]
-        k = split(pi.rng.rng.key)
+        k = split(state(pi.rng.rng))
         Random.seed!(pi.rng, k[1])
     end
     return nothing
 end
 
+"""
+    seed_from_rng!(pc::ParticleContainer, rng::Random.AbstractRNG, ref::Union{Particle,Nothing}=nothing)
+
+Set seeds of particle rng from user-provided `rng`
+"""
+function seed_from_rng!(
+    pc::ParticleContainer{T,<:TracedRNG{R,N,<:Random123.AbstractR123{I}}},
+    rng::Random.AbstractRNG,
+    ref::Union{Particle,Nothing}=nothing,
+) where {T,R,N,I}
+    n = length(pc.vals)
+    nseeds = isnothing(ref) ? n : n - 1
+    rngs = vcat([pc[i].rng for i in 1:nseeds], [pc.rng])
```

Comment:
This allocation could be avoided by splitting the loop.

Reply:
Yes, I've changed it, we don't really need to collect these